vscode-dotty/src/worksheet.ts

```diff
@@ -214,12 +215,13 @@ export class WorksheetProvider implements Disposable {
         const worksheet = this.worksheetFor(event.document)
         if (worksheet) {
           // Block file saving until the worksheet is ready to be run.
-          worksheet.prepareForRunning()
+          worksheet.prepareRun()
```

Comment:
I don't think it's going to block unless we return a `Promise` here

Reply:
It's actually trickier than just returning the Thenable, we need to use waitUntil from https://code.visualstudio.com/docs/extensionAPI/vscode-api#TextDocumentWillSaveEvent

Reply:
(This is done in the latest commit)